packages/custom-question-api/src/index.ts

```diff
@@ -1,7 +1,8 @@
 import { AnswerSemanticType, CustomQuestionMessage } from '@formsort/constants';
 import { getValueFromWindowParent, sendMessageToWindowParent } from './utils';
 
-type AnswerPrimitiveType = number | string | boolean;
+// eslint-disable-next-line @typescript-eslint/ban-types
+type AnswerPrimitiveType = number | string | boolean | object;
```

Comment:
`object` means anything except for primitive types, i.e. it can be array or function. I think the intention here is not include functions and arrays. [ref](https://exploringjs.com/tackling-ts/ch_typing-objects.html#:~:text=In%20TypeScript%2C%20object%20is%20the,any%20properties%20of%20a%20value.) So I'd recommend using `Record<string, unknown>` 
<img width="144" alt="image" src="https://user-images.githubusercontent.com/7889565/154464801-ee5c9bc8-1a86-4f0a-a12c-f39007d3ebbe.png">

Reply:
I agree that `object` is too permissive, but the difficulty with `Record<string, unknown>` is that it can't be used with a type param:
```ts
type AnswerPrimitiveType = number | string | boolean | Record<string, unknown>;
type AnswerType = AnswerPrimitiveType | AnswerPrimitiveType[];
...

export const getAnswerValue = <T extends AnswerType = AnswerType>() =>
  getValueFromWindowParent<T | undefined>(
    CustomQuestionMessage.REQUEST_ANSWER_VALUE_MSG,
    CustomQuestionMessage.SET_ANSWER_VALUE_MSG
  );

interface Answer {
  name: string;
}

async function test() {
  const val = getAnswerValue<Answer>(); // <-- TS Error... Index signature for type 'string' is missing in type 'Answer'.ts(2344)
}
```
Also made a [playground link](https://www.typescriptlang.org/play?#code/C4TwDgpgBAggdgZwO4QE4AVUEsC2XhYBuEAKuNALxRwCuOARmlAD5QLDZwDmLU9A9vwA2EAIZxeAJQgBjfqgAmAHnacuAGig04Aazj8kcAHwBuAFChIsRClRkrVeMjSZc+IqXK8nt13gLE9hAA2gC65mYKskKiqNAAZtoyBPwSXBDAPmgAaqJCNNBKJFAQAB7AEHAKCNbOdl6ONmhBRgAUAJQAXFCY-HgIEEW82lHxWHAQCkZQUGZm4xWo8aIy0FmoUADeZjNwojgQ3arjXOYAvnOiCCBwMlCJtykSFewdWztQcojAUIR5UFR0pkmqhcvlBus2u1zkA) to illustrate.
I did a bit of research into it yesterday, but couldn't find an easy solution to allow the above.